### PR TITLE
Expose waiting for upload/download on SyncSession

### PR DESCRIPTION
**Pull request by jsflax**

### CODE REVIEW

File: RealmSwift/Sync.swift [hunk 1]
```diff
@@ -923,6 +923,34 @@ public extension SyncSession {
...
+            }
+        case .download:
+            __waitForDownloadCompletion(on: nil) { error in
+                block(error)
+            }
+        }
+    }
+    
+    @available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+    @_unsafeInheritExecutor
```
Comment (tgoyne) [inline RealmSwift/Sync.swift:942]: This shouldn't need `@_unsafeInheritExecutor` since SyncSession is Sendable.

File: RealmSwift/Sync.swift [hunk 2]
```diff
@@ -923,6 +923,34 @@ public extension SyncSession {
                                                 block(Progress(transferred: transferred, transferrable: transferrable))
         }
     }
+    
+    @preconcurrency
```
Comment (tgoyne) [inline RealmSwift/Sync.swift:927]: `@preconcurrency` doesn't make sense here.

File: RealmSwift/Sync.swift [hunk 3]
```diff
@@ -923,6 +923,34 @@ public extension SyncSession {
...
+    func wait(for direction: ProgressDirection, block: @Sendable @escaping (Error?) -> Void) {
```
Comment (tgoyne) [inline RealmSwift/Sync.swift:928]: This should expose the queue parameter for non-async usage.

**Review by tgoyne — COMMENTED** [review]
Needs a changelog entry.

**Comment by tgoyne** [comment]
Also if our idea is that people should mostly not be using these and that it's potentially a bad idea, the docs really should say that.

**Comment by jsflax** [comment]
I'll leave a warning and mark it in the changelog.

**Review by dianaafanador3 — APPROVED** [review]
LGTM!, There are some listing issues to be resolved before merging this

**Pull request merged**